tests/api_app/test_config_parse.py

```diff
@@ -0,0 +1,74 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+import random
+from hashlib import md5
+from unittest.mock import Mock, patch
+
+from django.test import TestCase
+
+from api_app.core.serializers import AbstractConfigSerializer
+
+
+class ConfigParseTests(TestCase):
+    @patch("api_app.core.serializers.AbstractConfigSerializer._md5_config_file")
+    @patch("api_app.core.serializers.AbstractConfigSerializer._read_config")
+    def test_extends(self, mock_read_config: Mock, mock_md5_config_file: Mock):
+        sample_analyzer_config = {
+            "Shodan_Honeyscore": {
+                "type": "observable",
+                "python_module": "shodan.Shodan",
+                "description": "scan an IP against Shodan Honeyscore API",
+                "disabled": False,
+                "external_service": True,
+                "leaks_info": False,
+                "observable_supported": ["ip"],
+                "config": {"soft_time_limit": 30, "queue": "default"},
+                "secrets": {
+                    "api_key_name": {
+                        "env_var_key": "SHODAN_KEY",
+                        "description": "",
+                        "required": True,
+                    }
+                },
+                "params": {
+                    "shodan_analysis": {
+                        "value": "honeyscore",
+                        "type": "str",
+                        "description": "",
+                    }
+                },
+            },
+            "Shodan_Search": {
+                "extends": "Shodan_Honeyscore",
+                "description": "scan an IP against Shodan Search API",
+                "params": {
+                    "shodan_analysis": {
+                        "value": "search",
+                        "type": "str",
+                        "description": "",
+                    }
+                },
+            },
+        }
+
+        pre_parsed_sample_config = {
+            "Shodan_Honeyscore": sample_analyzer_config["Shodan_Honeyscore"],
+            "Shodan_Search": {
+                **sample_analyzer_config["Shodan_Honeyscore"],
+                **sample_analyzer_config["Shodan_Search"],
+            },
+        }
+        pre_parsed_sample_config["Shodan_Search"].pop("extends")
+
+        mock_read_config.return_value = sample_analyzer_config
```

Comment:
The mock of _read_config (mock_read_config) should return the dict with the `extends` keyword, because the logic of replacement of the `extends` field with the "parent" analyzer's fields is done by your function (`_complete_config`) called inside `read_and_verify_config` after `_read_config`.

With `pre_parsed_sample_config` you don't test the logic, because the data are in the converted format in the mock of the file.

Reply:
Let's say we have 2 configs: 
```
A: {A.config},
B: {"extends": A, ...(B.config)}
```
and 
```
A: {A.config},
B: {...(A.config), ...(B.config)}
```
.
The goal of this PR is to ensure that these two configs are interpreted in the same manner. That's what I'm testing using `sample_analyzer_config` and `pre_parsed_sample_config`.

Reply:
@drosetti